airflow/contrib/operators/dataproc_operator.py

```diff
@@ -0,0 +1,433 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from airflow.contrib.hooks.gcp_dataproc_hook import DataProcHook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class DataProcPigOperator(BaseOperator):
+    """
+    Start a Pig query Job on a Cloud DataProc cluster.
+    """
+    template_fields = ['query', 'variables']
+    template_ext = ('.pg', '.pig',)
+    ui_color = '#0273d4'
+
+    @apply_defaults
+    def __init__(
+            self,
+            query,
+            variables=None,
+            dataproc_cluster='cluster-1',
+            dataproc_pig_properties=None,
```

Comment:
> I don't understand this question: "Same question for other _properties params as well."?
 
Disregard. You answered my question here:

> The dataproc_* parameters don't make sense to template because you will in general initialise them as default_arguments.